### PR TITLE
Fixing corner cases with WKWebView and hybrid tests

### PR DESCRIPTION
**Pull request by bhariharan**
A bunch of tiny fixes. Should be pretty robust now.

### CODE REVIEW

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m [hunk 1]
```diff
@@ -84,10 +84,6 @@ - (void) setLoadToken:(WKWebView *) webView {
     [self stringByEvaluatingJavaScriptFromString:[NSString stringWithFormat:@"window.__cordovaLoadToken=%ld", (long) _curLoadToken] webView:webView];
 }
 
-- (NSString *) evalForCurrentURL:(WKWebView *) webView {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m]: Not required here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m [hunk 2]
```diff
@@ -96,6 +92,9 @@ - (void) pollForPageLoadStart:(WKWebView *) webView {
         [self log:SFLogLevelVerbose format:@"Polled for page load start. Result = YES!"];
         _state = STATE_IOS5_POLLING_FOR_LOAD_FINISH;
         [self setLoadToken:webView];
+        if ([_delegate respondsToSelector:@selector(webView:didStartProvisionalNavigation:)]) {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m]: I had forgotten to add the delegate callback here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m [hunk 3]
```diff
@@ -115,6 +114,9 @@ - (void) pollForPageLoadFinish:(WKWebView *) webView {
     if ([self isPageLoaded:webView]) {
         [self log:SFLogLevelVerbose format:@"Polled for page load finish. Result = YES!"];
         _state = STATE_IDLE;
+        if ([_delegate respondsToSelector:@selector(webView:didFinishNavigation:)]) {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m]: I had forgotten to add the delegate callback here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m [hunk 4]
```diff
@@ -146,19 +148,6 @@ - (void) webView:(WKWebView *) webView decidePolicyForNavigationAction:(WKNaviga
         NSURLRequest *request = navigationAction.request;
         BOOL isTopLevelNavigation = [request.URL isEqual:[request mainDocumentURL]];
         if (isTopLevelNavigation) {
-            if ([self request:request isEqualToRequestAfterStrippingFragments:request]) {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m]: This is an incorrect check.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m [hunk 5]
```diff
@@ -195,6 +184,10 @@ - (void) webView:(WKWebView *) webView decidePolicyForNavigationAction:(WKNaviga
     }
 }
 
+- (void) webView:(WKWebView *) webView didCommitNavigation:(WKNavigation *) navigation {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m]: This was the key to all the problems. ```WKNavigationDelegate``` has more callbacks than I care for, but this was causing ```gap://ready``` to not fire, thereby leaving the tests in a limbo.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 6]
```diff
@@ -154,13 +160,20 @@ - (id)initWithConfig:(SFHybridViewConfig *)viewConfig
...
-        [self.settings setCordovaSetting:@"SFWKWebViewEngine" forKey:@"CordovaWebViewEngine"];
     }
     return self;
 }
 
+- (UIView *)newCordovaViewWithFrame:(CGRect)bounds
+{
+
+    // Setting our WKWebView based plugin as the engine to be used.
+    [self.settings setCordovaSetting:@"SFWKWebViewEngine" forKey:@"CordovaWebViewEngine"];
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m]: This is the right place in the sequence of events to set this.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 7]
```diff
@@ -154,13 +160,20 @@ - (id)initWithConfig:(SFHybridViewConfig *)viewConfig
...
+    UIView *view = [super newCordovaViewWithFrame:bounds];
+    self.navWebViewDelegate = [[SFWKWebViewNavigationDelegate alloc] initWithEnginePlugin:self.webViewEngine];
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m]: Now, this one is interesting. In the new model, you can only have 1 navigation delegate, which means, it's either your view controller (in our case), or the stock navigation delegate. So, we need to invoke default behavior as well (the equivalent of ```super``` call previously). The reason it's done here is because we need to ensure our engine is initialized before we do this.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 8]
```diff
@@ -451,6 +464,13 @@ - (void)configureRemoteStartPage
     startPageConfigured = YES;
 }
 
+- (void) webView:(WKWebView *) webView didStartProvisionalNavigation:(WKNavigation *) navigation
+{
+    if (self.navWebViewDelegate) {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m]: Former ```super``` call.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 9]
```diff
@@ -501,6 +521,9 @@ - (void) webView:(WKWebView *) webView decidePolicyForNavigationAction:(WKNaviga
              }];
             decisionHandler(WKNavigationActionPolicyCancel);
         }
+        if (self.navWebViewDelegate) {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m]: Former ```super``` call.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 10]
```diff
@@ -530,6 +553,9 @@ - (void) webView:(WKWebView *) webView didFinishNavigation:(WKNavigation *) navi
                 _foundHomeUrl = YES;
             }
         }
+        if (self.navWebViewDelegate) {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m]: Former ```super``` call.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m (hunk 5 above)
Comment (huminzhi) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.m]: 👍
